docs/architecture.md

```diff
@@ -74,9 +74,17 @@ The Cortex services are:
 - [Ingester](#ingester)
 - [Querier](#querier)
 - [Query frontend](#query-frontend) (optional)
+- [Query scheduler](./operations/scalable-query-frontend.md#query-scheduler) (optional)
 - [Ruler](#ruler) (optional)
 - [Alertmanager](#alertmanager) (optional)
 - [Configs API](#configs-api) (optional)
+- [Compactor](./blocks-storage/compactor.md) (required for blocks storage)
```

Comment:
Just checking: is this strictly required for running blocks storage, or is it only required if using blocks storage *and* the bucket index?

Reply:
The compactor also dedupes if you have replication across ingesters, and bulks up the blocks from 2 hours to 24 hours which reduces the cost of historic queries substantially.  So you can technically run without it, but nobody should do.